src/ExtendedXmlSerializer/ExtensionModel/Content/Members/AllMembersParameterizedActivators.cs

```diff
@@ -0,0 +1,94 @@
+// MIT License
+//
+// Copyright (c) 2016-2018 Wojciech Nag�rski
+//                    Michael DeMond
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using ExtendedXmlSerializer.ContentModel.Members;
+using ExtendedXmlSerializer.Core;
+using ExtendedXmlSerializer.Core.Sources;
+using ExtendedXmlSerializer.ExtensionModel.Types;
+using ExtendedXmlSerializer.ReflectionModel;
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+
+namespace ExtendedXmlSerializer.ExtensionModel.Content.Members
+{
+	sealed class AllMembersParameterizedActivators : IActivators
+	{
+		readonly IActivators _activators;
+		readonly IQueriedConstructors _constructors;
+		readonly IMemberAccessors _accessors;
+        readonly ITypeMembers _typeMembers;
+        readonly IConstructorMembers _members;
+
+		public AllMembersParameterizedActivators(IActivators activators, IQueriedConstructors constructors,
+		                               IConstructorMembers members, IMemberAccessors accessors,
+                                       ITypeMembers typeMembers)
+		{
+			_activators = activators;
+			_constructors = constructors;
+			_members = members;
+			_accessors = accessors;
+            _typeMembers = typeMembers;
+        }
+
+		public IActivator Get(Type parameter)
+		{
+			var typeInfo = parameter.GetTypeInfo();
+			var constructor = _constructors.Get(typeInfo);
+			var members = constructor != null ? _members.Get(constructor) : null;
+            var typeMembers = _typeMembers.Get(typeInfo);
```

Comment:
Looks good.  Two issues here, however:

1. We want to only call `_typeMembers.Get` when members are found (`members != null` is true).
1. We want to remove all properties already found in the members.  Otherwise, properties in the constructor will get set twice.  This can be done with `typeMembers.Except(members.GetValueOrDefault())`.

Reply:
1. you are right
2. this is not necessary, only typeMembers are passed here so no duplicates in the array

Reply:
OK cool... last item.  Type members are passed in, but consider the case where a member is used in the constructor.  The constructor assigns it, and then the type members gets enumerated via the `ApplyMemberValuesCommand` to assign it again.

However, I did just now check it, and it seems that while members are _enumerated_ twice, they are only assigned once.

That is close enough for me.  I am going accept your work here and then clean up a few things and make a further commit.

Thanks again for your contribution!